### PR TITLE
chore(generator): add env var to reduce scope of generate-libraries build

### PR DESCRIPTION
**Pull request by scotthart**
Quality of life improvement for doing generator development.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10150)

### CODE REVIEW

File: ci/cloudbuild/build.sh [hunk 1]
```diff
@@ -191,6 +191,13 @@ export TRIGGER_TYPE
 export VERBOSE_FLAG
 export LOG_LINKER_PAT
 
+# Default to generating all libraries.
+if [ -n "${GENERATE_GOLDEN_ONLY+set}" ]; then
+  GENERATE_GOLDEN_ONLY=${GENERATE_GOLDEN_ONLY:-}
+else
+  GENERATE_GOLDEN_ONLY=""
+fi
```
Comment (devbww) [inline ci/cloudbuild/build.sh:199]: This seems to be the equivalent of `: ${GENERATE_GOLDEN_ONLY=}`, which can be read as "if GENERATE_GOLDEN_ONLY is unset, then set it to the empty string".

But, you could just use `${GENERATE_GOLDEN_ONLY-}` below instead.

File: ci/cloudbuild/builds/generate-libraries.sh [hunk 2]
```diff
@@ -32,22 +32,39 @@ bazel run --action_env=GOOGLE_CLOUD_CPP_ENABLE_CLOG=yes \
...
+fi
+
+if [ -z "${GENERATE_GOLDEN_ONLY}" ]; then
+  # TODO(#5821): The generator should run clang-format on its output files itself
+  # so we don't need this extra step.
+  io::log_h2 "Formatting generated code"
+  git ls-files -z -- '*.h' '*.cc' |
+    xargs -P "$(nproc)" -n 1 -0 clang-format -i
+else
+  io::log_red "Only formatting generator golden code."
```
Comment (devbww) [inline ci/cloudbuild/builds/generate-libraries.sh:54]: s/generator/generated/

File: ci/cloudbuild/builds/generate-libraries.sh [hunk 3]
```diff
@@ -32,22 +32,39 @@ bazel run --action_env=GOOGLE_CLOUD_CPP_ENABLE_CLOG=yes \
...
+    'generator/integration_tests/golden/internal/*.cc' \
+    'generator/integration_tests/golden/*.cc' |
+    xargs -P "$(nproc)" -n 1 -0 clang-format -i
+fi
+
+if [ -z "${GENERATE_GOLDEN_ONLY}" ]; then
+  io::log_h2 "Updating protobuf lists/deps"
+  external/googleapis/update_libraries.sh
+else
+  io::log_red "Skipping update of generated libraries dependencies."
```
Comment (devbww) [inline ci/cloudbuild/builds/generate-libraries.sh:66]: "Skipping update of protobuf lists/deps" perhaps, to match non-skipping message.

File: ci/cloudbuild/cloudbuild.yaml [hunk 4]
```diff
@@ -25,6 +25,7 @@ options:
     'COMMIT_SHA=${COMMIT_SHA}',
     'PR_NUMBER=${_PR_NUMBER}',
     'TRIGGER_TYPE=${_TRIGGER_TYPE}',
+    'GENERATE_GOLDEN_ONLY=""',
```
Comment (devbww) [inline ci/cloudbuild/cloudbuild.yaml:28]: Is this important, given that the above code doesn't distinguish between unset and empty?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10150)* status: 0 of 3 files reviewed, 4 unresolved discussions (waiting on @devbww)

___
*[`ci/cloudbuild/build.sh` line 199 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10150#-NFjitGAYhefRfcY_7E_-r2-199:-NFnwjRB6C5x3phANrVS:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/044645c5d7fbe32fce486264ebe2e15291f36347/ci/cloudbuild/build.sh#L199)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

This seems to be the equivalent of `: ${GENERATE_GOLDEN_ONLY=}`, which can be read as "if GENERATE_GOLDEN_ONLY is unset, then set it to the empty string".

But, you could just use `${GENERATE_GOLDEN_ONLY-}` below instead.
</blockquote></details>

Done.
___
*[`ci/cloudbuild/cloudbuild.yaml` line 28 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10150#-NFk48C47o5RAaXGh21w-r2-28:-NFnwuuZ549_J2uKLoYe:b-8ebus3) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/044645c5d7fbe32fce486264ebe2e15291f36347/ci/cloudbuild/cloudbuild.yaml#L28)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Is this important, given that the above code doesn't distinguish between unset and empty?
</blockquote></details>

So the code in build.sh only passes the env var when using docker, for cloud build, this is the only way I can prevent getting the following error:
`Step #3: /workspace/ci/cloudbuild/builds/generate-libraries.sh: line 35: GENERATE_GOLDEN_ONLY: unbound variable`
___
*[`ci/cloudbuild/builds/generate-libraries.sh` line 54 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10150#-NFjitGAYhefRfcY_7Ea-r2-54:-NFnx8giAMLm6fn1f85S:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/044645c5d7fbe32fce486264ebe2e15291f36347/ci/cloudbuild/builds/generate-libraries.sh#L54)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

s/generator/generated/
</blockquote></details>

Done.
___
*[`ci/cloudbuild/builds/generate-libraries.sh` line 66 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10150#-NFjitGAYhefRfcY_7Ea-r2-66:-NFnx9FFBhmQn8lgJkiw:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/044645c5d7fbe32fce486264ebe2e15291f36347/ci/cloudbuild/builds/generate-libraries.sh#L66)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

"Skipping update of protobuf lists/deps" perhaps, to match non-skipping message.
</blockquote></details>

Done.

**Review by devbww — APPROVED** [review]

**Pull request merged**